agent/src/agent.ts

```
@@ -85,16 +86,24 @@ export class Agent extends MessageHandler {
     }
 
     private setClient(config: ConnectionConfiguration): void {
-        this.client = createClient({
-            editor: new AgentEditor(this),
-            config: { ...config, useContext: 'none' },
-            setMessageInProgress: messageInProgress => {
-                this.notify('chat/updateMessageInProgress', messageInProgress)
-            },
-            setTranscript: () => {
-                // Not supported yet by agent.
-            },
-            createCompletionsClient: config => new SourcegraphNodeCompletionsClient(config),
-        })
+        const graphqlClient = new SourcegraphGraphQLAPIClient(config)
+        const voided = graphqlClient
+            .isCodyEnabled()
+            .catch(error => console.error(error))
```

It would be better to make the method async and propagate the failure to the client, unless this is really exceptionally something we want to blackhole in this way.

My philosophy is it is better to propagate errors, track failures and drive those down. Locally swallowing errors works against that.

But again if this is a special case, that's OK... maybe a comment indicating why.